### DISCUSSION TITLE
DnD touch/mobile support & strange behaviour on desktop

### DISCUSSION THREAD

**Discussion by radulle** [KevinVandy/material-react-table#348] [Q&A]
It seems that DnD is not working on mobile. Are there plans for touch support?

In demos, when I drop item it goes back to it's original position and then table updates. It gives a feeling like drop was unsuccessful. Is there a way around it?

**Comment by radulle** [comment]
Same seems to be happening when I drag and drop columns.

**Comment by KevinVandy** [comment]
The current drag and drop features are not supported with most touch input. This is only the column resizing and column/row dragging features that are effected. If there is a lightweight way to add those features in for touch, I'd like to.

Last time I tried, I was going to have to add an extra dependency that was at least 12kb, so decided it was not worth that compromise.

**Reply by radulle** [reply]
Yes, DnD libs are huge. Might there be a way to have DnD features as a plugin so that it does not affect everyone?

What do you think is the reason that DnD content is being moved back to it's origin on every drop?

**Reply by radulle** [reply]
Libs like dnd-kit are very nice but 10k+. Maybe on mobile DnD can be substituted by using a dropdown menu (move left, right, up, down)?

**Reply by KevinVandy** [reply]
I'm certainly open to adding this. DnD-kit is what I would use. So far it seems that not having dragging features on mobile is a fine tradeoff. I'll add more in the documentation for its current state though.
